plugin/trino-loki/src/main/java/io/trino/plugin/loki/LokiTableFunctionProvider.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.loki;
+
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import io.trino.plugin.base.classloader.ClassLoaderSafeConnectorTableFunction;
+import io.trino.spi.function.table.ConnectorTableFunction;
+
+import static java.util.Objects.requireNonNull;
+
+public class LokiTableFunctionProvider
+        implements Provider<ConnectorTableFunction>
+{
+    private final LokiMetadata lokiMetadata;
+
+    @Inject
+    public LokiTableFunctionProvider(LokiMetadata lokiMetadata)
+    {
+        this.lokiMetadata = requireNonNull(lokiMetadata, "lokimetadata is null");
+    }
+
+    @Override
+    public ConnectorTableFunction get()
+    {
+        return new ClassLoaderSafeConnectorTableFunction(new LokiTableFunction(this.lokiMetadata), getClass().getClassLoader());
```

Comment:
Can we use ClassLoaderSafe implementation for other components as well like Metadata, SplitManager, RecordCursor ?

Reply:
I do not know how to do the classes `LokiMetadata` etc doe not have a get method and the binding like in the [KafakaConnectorModule](https://github.com/trinodb/trino/blob/766ff04b4643a37729452c002a24c4f6b8f54b43/plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaConnectorModule.java#L55-L56) didn't work.